### PR TITLE
Pathways interface simplification

### PR DESCRIPTION
**Pull request by luisfabib**
See #39 
Simplified the interface for defining dipolar pathways in `dipolarkernel` and `dipolarbackground`. Instead of using a matrix for defining the pathways, using NaN for unmodulated pathways, these are now defined as a list of pathways, each being a list of parameters.

Refactored and simplified code for both functions profiting the new interface.

### CODE REVIEW

File: deerlab/dipolarbackground.py [hunk 1]
```diff
@@ -48,73 +50,58 @@ def dipolarbackground(t,pathinfo,Bmodel,renormalize = True,overtonecoeff = 1):
         lam = 0.4 # modulation depth main signal
         conc = 200   # spin concentration (uM)
 
-        pathinfo = [[],[]]
-        pathinfo[0] = [1-lam, NaN] # unmodulated part, gives offset
-        pathinfo[1] = [lam, 0] # main modulation, refocusing at time zero
+        pathinfo = []
```
Comment (stestoll) [inline deerlab/dipolarbackground.py:53]: Can you simplify this to `pathinfo = [[1-lam],[lam,0]]`? I don't think using `append` is the dominant use case.

File: deerlab/dipolarkernel.py [hunk 2]
```diff
@@ -81,8 +84,10 @@ def dipolarkernel(t,r,pathinfo = 1, B = 1, method = 'fresnel', excbandwidth = in
     To specify the standard model for 4-pulse DEER with an unmodulated offset and a single dipolar pathway that refocuses at time 0, use::
     
         lam = 0.4  # modulation depth main signal
-        pathinfo = [[1-lam, nan], [lam, 0]]
-
+        pathinfo = []
+        pathinfo.append([1-lam])      # unmodulated part, gives offset
+        pathinfo.append([lam, 0])     # main modulation, refocusing at time zero
+
```
Comment (stestoll) [inline deerlab/dipolarkernel.py:90]: Again, probably better `pathinfo = [[1-lam],[lam,0]]`.

File: deerlab/dipolarkernel.py [hunk 3]
```diff
@@ -97,22 +102,22 @@ def dipolarkernel(t,r,pathinfo = 1, B = 1, method = 'fresnel', excbandwidth = in
...
-        tau2 = 4                       # refocusing time of 2+1 contribution
-        pathinfo = [[],[],[]]
-        pathinfo[0] = [Lam0,   nan]    # unmodulated part, gives offset
-        pathinfo[1] = [lama,    0 ]    # main modulation, refocusing at time zero
-        pathinfo[2] = [lam21, tau2]    # 2+1 modulation, refocusing at time tau2
+        tau2 = 4                       # refocusing time (us) of 2+1 contribution
+        pathinfo = []
+        pathinfo.append([Lam0])           # unmodulated part, gives offset
+        pathinfo.append([lam1,   0  ])    # main modulation, refocusing at time zero
+        pathinfo.append([lam2,  tau2])    # 2+1 modulation, refocusing at time tau2
```
Comment (stestoll) [inline deerlab/dipolarkernel.py:110]: `pathway = [[Lam0],[lam1,0],[lam2,tau2]]`
Also, I think `pathway = [Lam0,[lam1,0],[lam2,tau2]]` should work for usability.

File: docsrc/source/basics.rst [hunk 4]
```diff
@@ -87,10 +87,9 @@ The returned output is
 
 .. code-block:: python
 
-    pathways =[[0.7,  NaN],
-               [0.3,   0 ]]
+    pathways =[[0.7], [0.3,   0 ]]
```
Comment (stestoll) [inline docsrc/source/basics.rst:90]: Make it possible to use `[0.7, [0,3.0]]`.

**Review by stestoll — CHANGES_REQUESTED** [review]

File: docsrc/source/basics.rst (hunk 4 above)
Comment (luisfabib) [reply docsrc/source/basics.rst:90]: It is already possible. I took care of that.

File: deerlab/dipolarbackground.py (hunk 1 above)
Comment (luisfabib) [reply deerlab/dipolarbackground.py:53]: The main idea was to define each pathway individually more clearly. But I agree that `append` might not be the most common use.

File: deerlab/dipolarkernel.py (hunk 3 above)
Comment (luisfabib) [reply deerlab/dipolarkernel.py:109]: See 0f8017d

File: deerlab/dipolarkernel.py (hunk 2 above)
Comment (luisfabib) [reply deerlab/dipolarkernel.py:88]: See 0f8017d

File: deerlab/dipolarbackground.py (hunk 1 above)
Comment (luisfabib) [reply deerlab/dipolarbackground.py:53]: See 0f8017d

**Review by stestoll — CHANGES_REQUESTED** [review]
There seem to be some inconsistencies regarding `pathinfo` vs `pathways` across several files. I do like `pathways`, it's more directly linked to the underlying concept. I think we should use it throughout.

**Comment by luisfabib** [comment]
Yes, I like `pathways` better too. The current keyword is `pathinfo` which might be confusing.

**Review by stestoll — APPROVED** [review]

**Pull request merged**